src/OpenLoco/Objects/ObjectManager.cpp

```diff
@@ -998,13 +1002,88 @@ namespace OpenLoco::ObjectManager
         throw std::runtime_error("Object not loaded at this index");
     }
 
+    bool computeObjectChecksum(const ObjectHeader& object, stdx::span<const uint8_t> data);
+
     // 0x00471BC5
     static bool load(const ObjectHeader& header, LoadedObjectId id)
     {
-        registers regs;
-        regs.ebp = X86Pointer(&header);
-        regs.ecx = static_cast<int32_t>(id);
-        return (call(0x00471BC5, regs) & X86_FLAG_CARRY) == 0;
+        // somewhat duplicates isObjectInstalled
+        const auto installedObjects = getAvailableObjects(header.getType());
+        auto res = std::find_if(std::begin(installedObjects), std::end(installedObjects), [&header](auto& obj) { return *obj.second._header == header; });
+        if (res == std::end(installedObjects))
+        {
+            // Object is not installed
+            return false;
+        }
+
+        const auto& installedObject = res->second;
+        const auto filePath = Environment::getPath(Environment::PathId::objects) / fs::u8path(installedObject._filename);
+
+        SawyerStreamReader stream(filePath);
+        ObjectHeader loadingHeader;
+        stream.read(&loadingHeader, sizeof(loadingHeader));
+        if (loadingHeader != header)
+        {
+            // Something wrong has happened and installed object does not match index
+            // Vanilla continued to search for subsequent matching installed headers.
+            return false;
+        }
+
+        // Vanilla would branch and perform more efficient readChunk if size was known from installedObject.ObjectHeader2
+        auto data = stream.readChunk();
+
+        if (!computeObjectChecksum(loadingHeader, data))
+        {
+            // Something wrong has happened and installed object checksum is broken
+            return false;
+        }
+
+        // Copy the object into Loco freeable memory (required for when load loads the object)
+        auto* object = reinterpret_cast<Object*>(malloc(data.size()));
+        std::copy(std::begin(data), std::end(data), reinterpret_cast<uint8_t*>(object));
+
+        if (!callObjectFunction(loadingHeader.getType(), *object, ObjectProcedure::validate))
+        {
+            free(object);
+            object = nullptr;
+            // Object failed validation
+            return false;
+        }
+
+        if (_totalNumImages >= Gfx::G1ExpectedCount::kObjects + Gfx::G1ExpectedCount::kDisc)
+        {
+            free(object);
+            object = nullptr;
+            // Too many objects loaded and no free image space
+            return false;
+        }
+
+        // sub_471BCE only code
```

Comment:
do we need these comments? also they haven't been formatted by clangformat, the whitespaces are a bit off

Reply:
I was going to do a follow on pr that makes this code active.